include/irr/asset/IIncluder.h

```diff
@@ -0,0 +1,45 @@
+#ifndef __IRR_I_INCLUDER_H_INCLUDED__
+#define __IRR_I_INCLUDER_H_INCLUDED__
+
+#include "irr/core/IReferenceCounted.h"
+#include "irr/core/Types.h"
+#include "IFileSystem.h"
+#include <string>
+
+namespace irr { namespace asset
+{
+
+class IIncluder : public core::IReferenceCounted
+{
+protected:
+    core::vector<std::string> m_searchDirectories;
+
+    virtual ~IIncluder() = default;
+
+public:
+    IIncluder() : m_searchDirectories{""} {}
+
+    virtual void addSearchDirectory(const std::string& _searchDir) { m_searchDirectories.push_back(_searchDir); }
+
+    std::string getInclude(const std::string& _path) const
+    {
+        for (const std::string& searchDir : m_searchDirectories)
```

Comment:
order in which you try them is important, you dont want to try a parent directory before a deeper (more matching) one